core-chart/templates/argocd/setup.yaml

```diff
@@ -4,117 +4,57 @@ kind: ConfigMap
 metadata:
   name: "{{ .Release.Name }}-argocd-kubestellar-setup"
 data:
-  # This script is used by the "{{ .Release.Name }}-argocd-kubestellar-setup" job to
-  # 1. register KubeStellar WDSes as Argo CD target clusters
-  # 2. create any Argo CD application listed in KubeStelar values
+  # This script is used by the "{{ .Release.Name }}-argocd-kubestellar-setup" job to:
+  # 1. Register KubeStellar WDSes as Argo CD target clusters
+  # 2. Create any Argo CD applications listed in KubeStellar values
+  # Note: Waiting logic for dependencies moved to initContainers for better reliability
   argocd-setup.sh: |
     #!/usr/bin/env bash
     set -e
 
-    WAIT_BEFORE_RETRY=5
     NS={{ .Release.Namespace }}
+    echo "[$(date +"%Y-%m-%d %H:%M:%S")] Getting ArgoCD pod and credentials..."

```

Comment:
I think that the time zone is much more interesting than the date (which will be clear from the context).

Reply:
@Rupam-It: yes, I think that would be an improvement.